qutebrowser/mainwindow/statusbar/command.py

```diff
@@ -54,14 +54,14 @@ class Command(misc.MinimalLineEditMixin, misc.CommandLineEdit):
     show_cmd = pyqtSignal()
     hide_cmd = pyqtSignal()
 
-    def __init__(self, win_id, parent=None):
-        misc.CommandLineEdit.__init__(self, parent)
+    def __init__(self, *, win_id, private, parent=None):
+        misc.CommandLineEdit.__init__(self, parent=parent)
         misc.MinimalLineEditMixin.__init__(self)
         self._win_id = win_id
-        command_history = objreg.get('command-history')
-        self.history.handle_private_mode = True
-        self.history.history = command_history.data
-        self.history.changed.connect(command_history.changed)
+        if not private:
+            command_history = objreg.get('command-history')
+            self.history.history = command_history.data
+            self.history.changed.connect(command_history.changed)
```

Comment:
This is probably a super stretch goal, but it might be nice to include browser history from non-private tab (read only) as well, so you can continue to use the 'history completion' to navigate to sites previously used. I'm not sure how many people would want that though.